test/scripts/generate-react-native-app.sh

```diff
@@ -16,5 +16,8 @@ else
   --defaults --no-insight --skip-git
 fi
 
+# pin specific version of expo
+npm i expo@48.0.6
+
```

Comment:
There's an issue in the latest `expo`, which mean `e2e` will fails in the local env as well:

<img width="1082" alt="Screenshot 2023-12-03 at 10 44 05" src="https://github.com/jhipster/generator-jhipster-react-native/assets/31362988/c3e1b03b-5008-4785-9db2-58619f61694f">

https://github.com/expo/expo/issues/25551

Reply:
But it failed:

<img width="583" alt="Screenshot 2023-12-03 at 19 47 48" src="https://github.com/jhipster/generator-jhipster-react-native/assets/31362988/fd4ad08f-850a-4a57-ad56-8121f18e70b6">